spi/data-plane/data-plane-spi/src/main/java/org/eclipse/edc/connector/dataplane/spi/iam/DataPlaneAccessTokenService.java

```diff
@@ -51,4 +51,14 @@ public interface DataPlaneAccessTokenService {
      * @return A {@link AccessTokenData} that contains the original claims and the data resource ({@link DataAddress}. If the token could not be restored, a failure is returned.
      */
     Result<AccessTokenData> resolve(String token);
+
+    /**
+     * Revokes the {@link AccessTokenData} associated with the id
+     *
+     * @param transferProcessId The id of the {@link AccessTokenData}
+     * @param reason            The reason for the revocation
+     * @return Success if revoked, failure otherwise
+     */
+    Result<Void> revoke(String transferProcessId, String reason);
```

Comment:
is the reason needed here? It makes sense in the `TransferProcess` context because the counter part then knows why, but I'm not sure the data plane needs to know. In fact in the implementation is not used

Reply:
For our internal impl we don't use it, but that might be useful for other impl? that's why i left it